README.rst

```diff
@@ -15,125 +15,137 @@ About
+Let us test whether everything works well locally in our containerised
+environment. Note how we mount our local directories ``inputs``, ``code`` and
+``outputs`` into the containerised environment:
 
 .. code-block:: console
 
-    $ docker push helloworld
+    $ rm -rf outputs && mkdir outputs
+    $ docker run -i -t  --rm \
+                -v `pwd`/code:/code \
+                -v `pwd`/inputs:/inputs \
+                -v `pwd`/outputs:/outputs \
+                python:2.7 \
+             python /code/helloworld.py --sleeptime 0
+    $ tail -1 outputs/greetings.txt
+    Hello Jane Doe!
 
-Workflow
-========
+4. Analysis workflow
+====================
 
 The hello world example is simple enough in that it does not require any complex
 workflow steps to run. It consists of running a single command only.
 Nevertheless, we shall demonstrate on how one could use the `Yadage
 <https://github.com/diana-hep/yadage>`_ workflow engine to represent the hello
 world application and its input and output arguments in a structured YAML
-manner:
-
-- `helloworld.yaml <helloworld.yaml>`_
-
-For example:
-
-.. code-block:: console
-
-   $ cat helloworld.yaml
-   stages:
-     - name: helloworld
-       dependencies: [init]
-       scheduler:
-         scheduler_type: 'singlestep-stage'
-         parameters:
-           name: {stages: init, output: name, unwrap: True}
-           delay: {stages: init, output: delay, unwrap: True}
-           outputfile: '{workdir}/greetings.txt'
-         step:
-           process:
-             process_type: 'string-interpolated-cmd'
-             cmd: 'python helloworld.py --name "{name}" --sleeptime {delay} --outputfile "{outputfile}"'
-           publisher:
-             publisher_type: 'frompar-pub'
-             outputmap:
-               outputfile: outputfile
-           environment:
-             environment_type: 'docker-encapsulated'
-             image: 'reanahub/reana-demo-helloworld'
-
-This provides a fully described "hello world" application that can be run on the
-REANA cloud.
+manner.
+
+- `workflow.yaml <workflow/yadage/workflow.yaml>`_
+
+**FIXME**
+
+REANA file
+==========
+
+Putting all together, we can describe our example hello world application, its
+runtime environment, the inputs, the code, the workflow and its outputs by means
+of the following REANA file:
+
+.. code-block:: yaml
+
+    version: 0.1.0
+    metadata:
+      - authors:
+        - Harri Hirvonsalo <hjhsalo@gmail.com>
+        - Diego Rodriguez <diego.rodriguez@cern.ch>
+        - Tibor Simko <tibor.simko@cern.ch>
+      - title: Hello world - A simple reusable analysis example
+      - date: 18 January 2017
+      - repository: https://github.com/reanahub/reana-demo-helloworld/
+    code:
+      - files:
+        - code/helloworld.py
+    inputs:
+      - files:
+        - inputs/names.txt
+      - parameters:
+        - sleeptime: 2
+    outputs:
+      - files:
+        - outputs/greetings.txt
+    environment:
```

Comment:
@diegodelemos @hjhsalo Shall we go aleady for the plural option?

```yaml
environments:
  - name: python
    type: docker
    image: python:2.7
  - name: fitting
    type: docker
    image: johndoe/some-compiled-environment
  - name: plotting
    type: docker
    image: reanahub/reana-env-root6
```

(`name` is optional, could probably leave it out... should be enough)

Reply:
Sure, but just as a disclaimer, I don't know enough about yadage to say how much work to support this needs or if this works out-of-the-box

Reply:
Yes, let’s do it now

Reply:
@hjhsalo Yadage won't see this, and the environments will be written in its yaml file directly anyway.  But `reana-client` might need some amendments later...